File: downloader.php
<?php
$id=$_SERVER["argv"][1];
if (!isset($id)) {
	exit("No ID");
} else if (!is_numeric($id)) {
	exit("ID not a number");
} else {
	echo "Download ID ".$id."\n";
}
$t=file_get_contents("http://www.6comic.com/comic/readmanga_".$id.".html?ch=1-1");
$t=iconv("BIG5", "UTF-8", $t);
preg_match("/var chs=(\d+)/",$t, $m);
$ch1=1;
$ch2=$m[1];
if (isset($_SERVER["argv"][2])) {
	if (preg_match("/^(\d+)-(\d+)$/", $_SERVER["argv"][2], $m)) {
		$ch1=$m[1];
		$ch2=$m[2];
	} else if (is_numeric($_SERVER["argv"][2])) {
		$ch1=$ch2=$_SERVER["argv"][2];
	} else if (!is_numeric($id)) {
		exit("Chapter not a number");
	}
	echo "Download Chapter ".$ch1."-".$ch2."\n";
} else {
	echo "Download All Chapters\n";
}
$pg1=1;
$pg2=PHP_INT_MAX;
if (isset($_SERVER["argv"][3])) {
	if (preg_match("/^(\d+)-(\d+)$/", $_SERVER["argv"][3], $m)) {
		$pg1=$m[1];
		$pg2=$m[2];
	} else if (is_numeric($_SERVER["argv"][3])) {
		$pg1=$pg2=$_SERVER["argv"][3];
	} else if (!is_numeric($id)) {
		exit("Page not a number");
	}
	echo "Download Page ".$pg1."-".$pg2."\n";
} else {
	echo "Download All Pages\n";
}
preg_match("/var cs='(.+?)';/", $t, $m);
$code=$m[1];
$hash=substr($code, 0, 3);
@mkdir("downloads");
@mkdir("downloads/".$id);
for ($i=$ch1; $i <= $ch2; $i++) {
	$prestr=$hash.$i;
	$prestr=substr($prestr, strlen($prestr)-4, 4);
	$startid=strpos($code, $prestr);
	if ($startid === false) {
		echo "Chapter ".$i." not found.\n";
		continue;
	}
	@mkdir("downloads/".$id."/".$i);
	$domain=substr($code, $startid+5, 1);
	$folder=substr($code, $startid+6, 1);
	preg_match("/(\d+)$/", substr($code, $startid+7, 3), $m);
	$page=$m[1];
	echo "Download Chapter ".$i." Page ".max($pg1,1)."-".min($pg2,$page)."\n";
	for ($j=max($pg1,1); $j <= min($pg2,$page); $j++) {
		echo "Downloading Chapter ".$i." Page ".$j."\n";
		$k=($j-1)%100+1;
		$startid2=$startid+10+($k-1)%10*3+floor(($k-1)/10);
		$imghash=substr($code, $startid2, 3);
		$url="http://img".$domain.".6comic.com:99/".$folder."/".$id."/".$i."/".str_pad($j, 3, "0", STR_PAD_LEFT)."_".$imghash.".jpg";
		$img=file_get_contents($url);
		file_put_contents("downloads/".$id."/".$i."/".$j.".jpg", $img);
	}
}
?>

File: downloader.py
import io
import os
import urllib.request
import re
import sys
import math
if len(sys.argv)<2:
	sys.exit("no id")
else:
	cid=sys.argv[1]
if cid.isdigit()==False:
	sys.exit("id not a number")
t=urllib.request.urlopen("http://www.6comic.com/comic/readmanga_"+cid+".html?ch=1-1").read().decode('BIG5')
cname = re.search('<title>(.+?)</title>', t)
if cname is None:
	sys.exit('comic name is not found')
else:
	cname = str.rstrip(cname.group(1).split('-')[0])[:-2]
	cname = re.sub('\\|\/|\:|\*|\?|\"|<|>|\|', ' ', cname)	# Fix filename incorrect
m=re.search("var chs=(\d+)",t)
ch1=1
ch2=m.group(1)
if len(sys.argv)>=3:
	m=re.search("^(\d+)-(\d+)$",sys.argv[2])
	if m!=None:
		ch1=m.group(1)
		ch2=m.group(2)
	elif sys.argv[2].isdigit():
		ch1=ch2=sys.argv[2]
	else:
		sys.exit("ch not a number")
print("download id="+cname+" ch="+str(ch1)+"-"+str(ch2))
m=re.search("var cs='(.+?)';",t)
code=m.group(1)
hashc=code[0:3]
if os.path.isdir("downloads")==False:
	os.mkdir("downloads")
if os.path.isdir("downloads/"+cname)==False:
	os.mkdir("downloads/"+cname)
for i in range(int(ch1),int(ch2)+1):
	prestr=hashc+str(i)
	prestr=prestr[-4:]
	startid=code.find(prestr)
	if startid==-1:
		print("ch "+str(i)+" not found.")
		continue
	if os.path.isdir("downloads/"+cname+"/"+str(i))==False:
		os.mkdir("downloads/"+cname+"/"+str(i))
	domain=code[startid+5:startid+6]
	folder=code[startid+6:startid+7]
	m=re.search("(\d+)$",code[startid+7:startid+10])
	page=m.group(1)
	for j in range(1,int(page)+1):
		print("downloading ch="+str(i)+" page="+str(j))
		k=(j-1)%100+1;
		startid2=startid+10+(k-1)%10*3+math.floor((k-1)/10)
		imghash=code[startid2:startid2+3]
		url="http://img"+domain+".6comic.com:99/"+folder+"/"+cid+"/"+str(i)+"/"+(str(j).zfill(3))+"_"+imghash+".jpg"
		img=urllib.request.urlopen(url).read()
		f=open("downloads/"+cname+"/"+str(i)+"/"+str(j)+".jpg","wb")
		f.write(img)
		f.close()
